--- origsrc/gDesklets-0.36.1/libdesklets/system/Makefile.am	2008-03-19 18:00:58.000000000 -0500
+++ src/gDesklets-0.36.1/libdesklets/system/Makefile.am	2008-12-30 02:49:06.237500000 -0600
@@ -23,7 +23,7 @@
 	OpenBSD
 
 
-gtop_la_LDFLAGS = -module -avoid-version -as-needed
+gtop_la_LDFLAGS = -module -avoid-version -no-undefined -lpython$(PYTHON_VERSION)
 gtop_la_LIBADD = $(GLIBTOP_LIBS)
 gtop_la_SOURCES = gtop.c
 
--- origsrc/gDesklets-0.36.1/utils/Makefile.am	2008-03-19 18:00:59.000000000 -0500
+++ src/gDesklets-0.36.1/utils/Makefile.am	2008-12-30 02:48:53.628125000 -0600
@@ -41,15 +41,15 @@
 
 svg_la_SOURCES = svg.c utils.c utils.h
 svg_la_LIBADD = $(SVG_LIBS)
-svg_la_LDFLAGS = -module -avoid-version -as-needed
+svg_la_LDFLAGS = -module -avoid-version -no-undefined -lpython$(PYTHON_VERSION)
 
 tiling_la_SOURCES = tiling.c render.c render.h utils.c utils.h
 tiling_la_LIBADD = $(TILING_LIBS)
-tiling_la_LDFLAGS = -module -avoid-version -as-needed
+tiling_la_LDFLAGS = -module -avoid-version -no-undefined -lpython$(PYTHON_VERSION)
 
 x11_la_SOURCES = x11.c utils.c utils.h
 x11_la_LIBADD = $(X11_LIBS)
-x11_la_LDFLAGS = -module -avoid-version -as-needed
+x11_la_LDFLAGS = -module -avoid-version -no-undefined -lpython$(PYTHON_VERSION)
 
 EXTRA_DIST = $(install_DATA)
 
