--- origsrc/gDesklets-0.36.1/libdesklets/system/gtop.c	2008-03-19 18:00:58.000000000 -0500
+++ src/gDesklets-0.36.1/libdesklets/system/gtop.c	2008-12-30 03:01:43.753125000 -0600
@@ -707,7 +707,9 @@
     glibtop_netload buf;
     union {
         char ip4[INET_ADDRSTRLEN];
+#ifdef AF_INET6
         char ip6[INET6_ADDRSTRLEN];
+#endif
         char hw [32]; /* 8 * 2 + 7 * 1 ... 32 is enough :) */
     } addr;
     const char *iface;
@@ -738,6 +740,7 @@
     my_dict_add_and_decref (d, "errors_out",         PyL_ULL (buf.errors_out));
     my_dict_add_and_decref (d, "errors_total",       PyL_ULL (buf.errors_total));
     my_dict_add_and_decref (d, "collisions",         PyL_ULL (buf.collisions));
+#ifdef AF_INET6
     my_dict_add_and_decref (d, "prefix6",
                             PyS_S (inet_ntop(AF_INET6, buf.prefix6, addr.ip6,
                                             sizeof addr.ip6)));
@@ -745,6 +748,7 @@
                             PyS_S (inet_ntop(AF_INET6, buf.address6, addr.ip6,
                                              sizeof addr.ip6)));
     my_dict_add_and_decref (d, "scope6", PyI_L(buf.scope6));
+#endif
 
     my_dict_add_and_decref (d, "hwaddress",
                            PyS_S (hwaddress_format_for_display(&buf, addr.hw,
